src/vivarium/framework/results/observation.py

```diff
@@ -96,6 +113,59 @@ def __init__(
         self.aggregator_sources = aggregator_sources
         self.aggregator = aggregator
 
+    @staticmethod
+    def initialize_results(
+        measure: str,
+        requested_stratification_names: List[str],
+        registered_stratifications: List[Stratification],
+        registered_stratification_names: Set[str],
+        missing_stratifications: Dict[str, Set[str]],
+        unused_stratifications: Set[str],
+    ) -> Tuple[pd.DataFrame, Set[str]]:
+        """Initialize a dataframe of 0s with complete set of stratifications as the
+        index and return it along with any registered but unused stratifications
+        """
+        requested_stratification_names = set(requested_stratification_names)
```

Comment:
I know this is what the code was already doing, but it doesn't make sense to be checking if we have missing or extra stratifications here. This method should just be initializing the dataframe. We should manage this validation natively in the manager.